include/mbedtls/ssl_cache.h

```diff
@@ -102,6 +102,9 @@ void mbedtls_ssl_cache_init(mbedtls_ssl_cache_context *cache);
  * \param session_id_len  The length of \p session_id in bytes.
  * \param session         The address at which to store the session
  *                        associated with \p session_id, if present.
+ *
+ * \return                \c 0 on success.
+ * \return                A negative error code on failure.
```

Comment:
We should explicitly document `CACHE_ENTRY_NOT_FOUND` here, since it's an error that callers are likely to want to handle.